### DISCUSSION TITLE
What are workers for?

### DISCUSSION THREAD

**Discussion by ferferga** [goauthentik/authentik#1713] [Q&A]
What are workers for in docker-compose deployments? Are they only for backups and system tasks or also help to load balance?

Thank you very much!

**Comment by BeryJu** [comment]
Workers run the backup and other system tasks, but they also run a lot of other tasks which arent' shown in the Web UI, for example they run the policies on all events being created, they send Emails for email stages, they sync LDAP Sources, and a bunch of other things I'm forgetting

**Reply by ferferga** [reply]
Okay, thank you very much for your clarification!

Much appreciated!

**Comment by arigon** [comment]
Why has Authentik in Go been built on a task queue architecture? What is the idea and the experience behind this decision?
And what parts are (still?) in Python and what parts are in Go?

Modern Go applications don't need such a Python-style architecture to achieve scalability because you could just use scalable microservices or goroutines. Why is there a worker needed to send emails? Just call the emailservice to send an email without the overhead of a REDIS queue.
